### PR TITLE
refactor: resolved lint errors and tests coverage

### PR DESCRIPTION
**Pull request by MuhammadAli-Paysys**
# SPDX-License-Identifier: Apache-2.0

This pull request primarily focuses on code quality improvements and dependency updates across the backend, with a strong emphasis on formatting, maintainability, and minor logic clarifications. The changes include dependency upgrades, code style adjustments for readability, and some minor refactoring for consistency and clarity.

**Dependency Updates:**

* Upgraded `@tazama-lf/frms-coe-lib` to version `8.0.0-rc.5` and `@tazama-lf/frms-coe-startup-lib` to `3.0.2-rc.5` in `backend/package.json` to ensure the latest features and fixes are available.

**Code Quality and Formatting Improvements:**

* Reformatted and refactored `backend/src/interceptors/audit.interceptor.ts` for improved readability, including consistent indentation, clearer destructuring, and expanded multi-line expressions. No logic changes were made. [[1]](diffhunk://#diff-edd2a986cdc17e60a74111e7cc2a1a8608caf87e13ebd243288a0ecb51d421bdL1-R17) [[2]](diffhunk://#diff-edd2a986cdc17e60a74111e7cc2a1a8608caf87e13ebd243288a0ecb51d421bdL22-R56) [[3]](diffhunk://#diff-edd2a986cdc17e60a74111e7cc2a1a8608caf87e13ebd243288a0ecb51d421bdL58-R81) [[4]](diffhunk://#diff-edd2a986cdc17e60a74111e7cc2a1a8608caf87e13ebd243288a0ecb51d421bdL80-R103) [[5]](diffhunk://#diff-edd2a986cdc17e60a74111e7cc2a1a8608caf87e13ebd243288a0ecb51d421bdL92-R117) [[6]](diffhunk://#diff-edd2a986cdc17e60a74111e7cc2a1a8608caf87e13ebd243288a0ecb51d421bdL151-R210) [[7]](diffhunk://#diff-edd2a986cdc17e60a74111e7cc2a1a8608caf87e13ebd243288a0ecb51d421bdL209-R262) [[8]](diffhunk://#diff-edd2a986cdc17e60a74111e7cc2a1a8608caf87e13ebd243288a0ecb51d421bdL315-R372) [[9]](diffhunk://#diff-edd2a986cdc17e60a74111e7cc2a1a8608caf87e13ebd243288a0ecb51d421bdL330-R406)
* Updated type signatures in `backend/src/interceptors/serialize.interceptor.ts` to use `unknown` instead of `any` for improved type safety. [[1]](diffhunk://#diff-984aa636383ab3536162d78c08a72819a50642ebc6b1a04f8f987f65173da1d4L10-R10) [[2]](diffhunk://#diff-984aa636383ab3536162d78c08a72819a50642ebc6b1a04f8f987f65173da1d4L21-R21)
* Applied consistent multi-line formatting and removed unnecessary blank lines in several service files (`config.service.ts`, `job.service.ts`, `scheduler.service.ts`, `admin-service-client.service.ts`) to enhance code clarity and maintainability. [[1]](diffhunk://#diff-5c689873dd7efec17bf7e33d3eea897614ed8c44023e19dc37bdd41cccb5b956L769-R772) [[2]](diffhunk://#diff-54234b797575b2b714ddbf97ebd00222d66303f2a5bee14cb6ccd14ecec7f899L310-R311) [[3]](diffhunk://#diff-54234b797575b2b714ddbf97ebd00222d66303f2a5bee14cb6ccd14ecec7f899L573-R575) [[4]](diffhunk://#diff-4d6aa05987e7164e55cc74f1a6988fdd588e3a6a7a678cf6b544e7bfd267fa7eL59-R61) [[5]](diffhunk://#diff-4d6aa05987e7164e55cc74f1a6988fdd588e3a6a7a678cf6b544e7bfd267fa7eL174-R178) [[6]](diffhunk://#diff-4d6aa05987e7164e55cc74f1a6988fdd588e3a6a7a678cf6b544e7bfd267fa7eL222-R228) [[7]](diffhunk://#diff-4d6aa05987e7164e55cc74f1a6988fdd588e3a6a7a678cf6b544e7bfd267fa7eL415-R426) [[8]](diffhunk://#diff-e6d99197f3f1d5ad47701dfcf7c19780ca69c38676aebbd3982d35e06ff2e88eL304-R306) [[9]](diffhunk://#diff-e6d99197f3f1d5ad47701dfcf7c19780ca69c38676aebbd3982d35e06ff2e88eL991-R992) [[10]](diffhunk://#diff-e6d99197f3f1d5ad47701dfcf7c19780ca69c38676aebbd3982d35e06ff2e88eL1007-R1006)

**Tooling Adjustments:**

* Changed the pre-commit hook in `.husky/pre-commit` to use `npx lint-staged` instead of `npm test`, ensuring only staged files are linted and checked before commits, which can speed up the commit process.

No breaking changes or significant logic modifications are introduced in this PR; the focus is on modernization, maintainability, and developer experience. Therefore, resolving lint errors and making the test coverage to above 95%.

image

image

image

## How was it tested?
- [X] Locally
- [X] Development Environment
- [ ] Not needed, changes very basic
- [X] Husky successfully run
- [X] Unit tests passing and Documentation done

## Summary by CodeRabbit

* **New Features**
  * Improved form validation with clearer messages and automatic scroll-to-first-error in data-enrichment flows.

* **Bug Fixes**
  * Dialog/confirmation modals behave more reliably (close/disable/confirm flows).
  * Delimiter defaults to a comma when left empty.
  * Schedule loading and connection parsing handle more response shapes and edge cases.

* **Tests**
  * Significant expansion of test coverage across UI components and utilities.

* **Chores**
  * Updated dev tooling to run linting/formatting on pre-commit and bumped related packages.

[Review Change Stack](https://app.coderabbit.ai/change-stack/tazama-lf/connection-studio/pull/62?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: frontend/src/common/Tables/CustomTable/index.tsx [hunk 1]
```diff
@@ -93,7 +93,9 @@ const CustomTable: React.FC<CustomTableProps> = ({
             <Pagination
               page={pagination.page + 1}
               count={totalPages}
-              onChange={(_, newPage) => { pagination.setPage(newPage); }}
+              onChange={(_, newPage) => {
+                pagination.setPage(newPage);
+              }}
```
[the remark this answers is not in the document]
Comment (MuhammadAli-Paysys) [reply frontend/src/common/Tables/CustomTable/index.tsx:98]: Already handling this whenever the pagination variable is set.

**Review by MAdeel95 — APPROVED** [review]

**Review by Sandy-at-Tazama — APPROVED** [review]
